frontend/src/routes/MonthlyWorkingHours.jsx

```diff
@@ -30,14 +40,61 @@ const MonthlyWorkingHours = () => {
         }
         setData(Object.values(transformedData));
       });
+  };
+
+  const makeGraphData = () => {
+    axios
+      .get(
+        `http://localhost:3000/monthly-working-hours/${selectedYear}/${selectedYearsBack}`
+      )
+      .then((response) => {
+        const transformedData = {};
+        for (let i = 0; i < response.data.data.length; i++) {
+          const item = response.data.data[i];
+          if (!transformedData[item.year]) {
+            transformedData[item.year] = { year: item.year, total: 0 };
+          }
+          transformedData[item.year][item.month] = item.total_hours;
+          transformedData[item.year].total += item.total_hours;
+        }
+
+        const months = {};
+        const dataForGraph = [];
+
+        for (const year in transformedData) {
+          for (const month in transformedData[year]) {
+            if (!isNaN(month)) {
+              if (!months[month]) {
+                months[month] = {};
+              }
+              months[month][year] = transformedData[year][month];
+            }
+          }
+        }
+
+        for (const month in months) {
+          const entry = { month: parseInt(month) };
+          for (const year in transformedData) {
+            entry[year] = months[month][year] || 0;
+          }
+          dataForGraph.push(entry);
+        }
+        dataForGraph.sort((a, b) => a.month - b.month);
+
+        setDataGraph(dataForGraph);
```

Comment:
Why not do all this in the makeTableData axios get method call instead? That way we avoid fetching the data twice from the backend. This function can be then removed and makeTableData could be perhaps called fetchData instead. Or since there will be just one function call in applyFilters, the contents of the function could be moved directly to applyFilters.